cl/sentinel/gossip.go

```diff
@@ -328,14 +333,18 @@ func (sub *GossipSubscription) Listen() {
 					sctx, sub.cf = context.WithCancel(sub.ctx)
 					go sub.run(sctx, sub.sub, sub.sub.Topic())
 					sub.subscribed.Store(true)
+					sub.s.updateENROnSubscription(sub.gossip_topic.Name, true)
+					log.Info("[Gossip] Subscribed to topic", "topic", sub.gossip_topic.Name)
 				}
 			}
 		}
 	}()
 }
 
 func (sub *GossipSubscription) OverwriteSubscriptionExpiry(expiry time.Time) {
-	sub.expiration.Store(expiry)
+	if expiry.After(sub.expiration.Load().(time.Time)) {
```

Comment:
I feel we need to take care of nil value of `sub.expiration` ? or it must be non-nil?

Reply:
it is expected to be non-nil